pkg/ccl/pgcryptoccl/pgcryptocipherccl/cipher_test.go

```diff
@@ -48,6 +49,23 @@ func FuzzEncryptDecryptAES(f *testing.F) {
 	})
 }
 
+func FuzzEncryptDecryptAESNoPadding(f *testing.F) {
+	for _, tc := range cipherTestCases {
+		f.Add(tc.plaintext, tc.key, tc.iv)
+	}
+	f.Fuzz(func(t *testing.T, plaintext []byte, key []byte, iv []byte) {
+		ciphertext, err := pgcryptocipherccl.Encrypt(plaintext, key, iv, "aes/pad:none")
+		if plaintextLength := len(plaintext); plaintextLength%aes.BlockSize != 0 {
```

Comment:
Rather than discarding fuzzing inputs that don't match the BlockSize, could we `0` pad them as necessary?

Reply:
If the user specifies `/pad:none` then they need to ensure the data length is a multiple of the block size. This mode should really only be used if the user is handling the padding themselves.

I would argue that it is more valuable to validate that we are correctly returning an error rather than zero padding it ourselves to generate more valid inputs.

There is also `FuzzEncryptDecryptAES` above that tests the default padding mode, which is PKCS.